### ISSUE TITLE
Stub file not found for XXX since 1.1.22

### ISSUE THREAD

**Issue by hadim** [microsoft/pyright#524]
Since 1.1.22 I got a lot of errors about `Stub file not found for XXX`. I guess I can disable this in a setting file but I would like to have this option turned off globally.

Any reason why this behavior is default now?

**Comment by erictraut** [comment]
Yes, this was intentionally changed, but I forgot to mention it in the release notes. If a type stub is not present, Pyright will now generate a warning by default. It has always treated this as an error in "strict" mode. If you want to disable the warning for your workspace, you can do so through the pyrightconfig.json file.

A type checker like Pyright will provide much more value if you have type stubs available. Type stubs are increasingly available for download for popular libraries. If you can't find one, Pyright can attempt to generate a "first draft" type stub from the implementation. Just click on the code action in the warning (the lightbulb that appears when you mouse over the warning).

**Comment by hadim** [comment]
I would like to avoid using a `pyrightconfig.json` per project and have this option available globally in the VSCode settings. Would that be possible to add such an option?

**Comment by erictraut** [comment]
It's possible, but I'd prefer not to. I could be convinced if I receive sufficient feedback from other users.

**Comment by hadim** [comment]
This modification of the default behavior turned all my VSCode python files to yellow. I work on multiple repositories at the same time and I don't want to have a Pyright configuration file for each of them. A global configuration would be really useful.

**Comment by erictraut** [comment]
Out of curiosity, do you enable the "pyright.useLibraryCodeForTypes" option? If not, you get no type information, type completion, etc. for any external library, and the type checking in your own code is going to be very weak at best. That's why type stubs are so important. With this option enabled, I can see how you might derive sufficient value from Pyright even without type stubs. 

Here's what I'm thinking. Maybe I should base the default value of the "reportMissingTypeStubs" on whether "pyright.useLibraryCodeForTypes" is enabled. If it is, I can assume that the user isn't using type stubs and is happy with the type information that can be inferred directly from the library code. If "pyright.useLibraryCodeForTypes" is disabled (as it is by default), then "reportMissingTypeStubs" will generate a warning. What do you think?

**Comment by hadim** [comment]
I don't use `useLibraryCodeForTypes`. I tried but it generates too many errors on which I don't have control because the issue is upstream (or maybe sometime Pyright don't generate the correct type annotation?).

I also can't report all the errors upstream or here because that would require a huge amount of time. I try whenever I can.

I am aware that type checking is only done on my own code (+ std lib I guess) and it's already saving me a lot of time actually and working very well!

I think he would be good indeed to disable `reportMissingTypeStubs` whenever `useLibraryCodeForTypes` is disabled too.

**Comment by erictraut** [comment]
I think it's important to flag "type stubs missing" at least as a warning because that will display a code action that allows you to generate a type stub. Without this, you'll be oblivious to the fact that there's a gaping hole in your type checking. 

Like I said, I'd be willing to reconsider this decision if I get enough feedback from other users. I'm going to re-open the issue so it's visible, and we'll see what others say.

**Comment by hadim** [comment]
Ok, thanks for this.

Sorry to insist again but as of today automatic stubs generation is not working very well at least in my hand (unless the error is upstream of course).

Here is one example. I just run the action to generate a type stub for `numpy`. I have two errors using two very basic `numpy` functions:

```python
import numpy as np

seed: int = 55
np.random.seed(seed)  # 'seed' is not a known member of module

indices: list[int] = [55, 88, 78, 66]
np.random.permutation(indices)  # 'permutation' is not a known member of module
```

Maybe there is something I don't do correctly, if that's the case don't hesitate to tell me. If I am correct here, then I think `reportMissingTypeStubs` at least when `useLibraryCodeForTypes` is disabled.

I don't think I will use Pyright if it reports a ton of errors I cannot fix...

**Comment by bfelbo** [comment]
We're in the same boat as @hadim. Right now we've been forced to create a `pyrightconfig.json` per project even for ones, where typing is not a priority.

I've tried creating typings using the code action appearing in the lightbulb, but it doesn't work that well. For instance, we use psycopg2 extensively and after creating typings, pyright gives the following errors:
```
from psycopg2.extensions import adapt, register_adapter, AsIs
'adapt' is unknown import symbol
'AsIs' is unknown import symbol
```

**Comment by bfelbo** [comment]
I can see how enforcing strict typing can be very useful for big or critical codebases. However, for smaller codebases, it'd be amazing to get pyright type checking without lots of warnings/errors for 3rd party libraries.

**Comment by jakebailey** [comment]
Unless I'm mistaken, having `useLibraryCodeForTypes` enabled means that it won't report missing stubs, as they aren't needed anymore (the real code is used), hence the suggestion to make them effectively mutually exclusive by default.

With that in mind, disabling the warnings when `useLibraryCodeForTypes` is not enabled is equivalent to the preference that the warning be disabled by default (your original expressed preference).

`numpy` is a fickle module that requires analysis of compiled code, something that pyright doesn't do during its stub generation (among some other odd edge cases I've experienced too and need to report). It's possible the same is happening for `psycopg2` as it's a wrapper around `libpq`, a C library.

**Comment by EthanC** [comment]
+1 for a global preference for `reportMissingTypeStubs`, per project configuration isn't the most favorable solution.

In my case, Pyright isn't even offering to generate stubs for [praw](https://github.com/praw-dev/praw) or [coloredlogs](https://github.com/xolox/python-coloredlogs), so I'm essentially left with warnings that I have no control over... Unless I decide to write type stubs for somebody else's libraries.

**Comment by erictraut** [comment]
OK, thanks for the feedback. I'll discuss with my colleagues about either reversing the behavior change or adding a global preference.

**Comment by erictraut** [comment]
I'm going to revert the behavior change for now. This will be in the next published version of Pyright.

**Comment by erictraut** [comment]
This is included in 1.1.23, which I just published. Thanks again for the feedback, and apologies for any inconvenience this caused.
